### DISCUSSION TITLE
Page data not updating when using router.push("/route")

### DISCUSSION THREAD

**Discussion by shellking4** [vercel/next.js#37361] [Help] [answered]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

Linux/Ubuntu 20.04

### Describe the Bug

As stated in the doc the getServerSideProps method gets called in these cases

- When you request this page directly, getServerSideProps runs at request time, and this page will be pre-rendered with the returned props
- When you request this page on client-side page transitions through [next/link](https://nextjs.org/docs/api-reference/next/link) or [next/router](https://nextjs.org/docs/api-reference/next/router), Next.js sends an API request to the server, which runs getServerSideProps.

But in my nextjs app it's not working that way. I have a form page to create new data after what I router.push to the data list page. The actual issue is that the data list is not showing the new entry. Why is that ? I even tried using router.replace(router.asPath) in the component I'm routing to but same thing. I'm wondering if the fact that I'm using next-pwa is the reason to this. Maybe the last request is cached or some stuff like that. Just wanna understand 

### Expected Behavior

GetServerSideProps should be called when I'm on the data list page and new added entry should show up automatically

### To Reproduce

I will soon provide a way to reproduce this issue.

**Comment by icyJoseph** [comment, marked as answer]
When you do router.push, in the devTools, can you see the  source of the request? if it's from the service worker it'd say so, and that would explain why the data is stale.

**Reply by shellking4** [reply]
Thank you very much for the reply @icyJoseph 
I will try what you said and see how it's behaving.

**Reply by shellking4** [reply]
> When you do router.push, in the devTools, can you see the source of the request? if it's from the service worker it'd say so, and that would explain why the data is stale.

I did inspect the request in the devTools and I noticed that it is being sent to this url  "/_next/data/6bfDGcm6KFnHlanmeKqjW/engines.json" and it's being initiated by workbox I think the response is what is cached.

**Reply by icyJoseph** [reply]
Right, so then the service working is doing the stale-while-revalidate strategy, https://developer.chrome.com/docs/workbox/modules/workbox-strategies/, which is most likely not what you want in this case. I wonder if next-pwa has a way to tell the worker to apply a different strategy? or perhaps apply them only to certain requests?

I guess in your case the `Network First (Network Falling Back to Cache)`, it's what's most appropriate.

Here you go: https://github.com/shadowwalker/next-pwa/issues/308

**Reply by shellking4** [reply]
Thank you very much I read the thread you sent (https://github.com/shadowwalker/next-pwa/issues/308) Very helpful but I don't actually know where to put the option for the handler thing. Here is my next.config.js

`
/** @type {import('next').NextConfig} */
const withPWA = require('next-pwa');
const runtimeCaching = require('next-pwa/cache');

module.exports = withPWA({
  pwa: {
    dest: 'public',
    runtimeCaching,
    register: true,
    skipWaiting: true,
    buildExcludes: [/middleware-manifest.json$/]
  },
  eslint: {
    ignoreDuringBuilds: true,
  },
});
How do I add it ?



`

**Reply by icyJoseph** [reply]
I think they either expect you to write your own, by copying their definition, or by mutating the `runtimeCaching` object, like so:

```js
/** @type {import('next').NextConfig} */
const withPWA = require('next-pwa');
const runtimeCaching = require('next-pwa/cache');

const nextDataIndex = runtimeCaching.findIndex((entry) => entry.options.cacheName === 'next-data');

if(nextDataIndex > 0){
  runtimeCaching[nextDataIndex].handler = 'NetworkFirst'
} else {
  throw new Error("Failed to find next-data object in runtime caching")
}

// now runtimeCaching has been mutated

module.exports = withPWA({
pwa: {
dest: 'public',
runtimeCaching,
register: true,
skipWaiting: true,
buildExcludes: [/middleware-manifest.json$/]
},
eslint: {
ignoreDuringBuilds: true,
},
});
```

Make sure to remove all application data from your browser, remove all service workers specifically, otherwise this fix might not be visible.

**Reply by shellking4** [reply]
😀😀😀 It worked fine.
 I updated the code, did what you suggested and deployed. It worked fine 🙂 I'm so glad. Thank you so much 🙏🙏🙏. Be blessed. All the same I would have to dig a little bit deeper in the service worker and caching stuff to understand more clearly. Thank you once again 🙂🙂🙂

**Reply by icyJoseph** [reply]
Awesome! Yeah, it is a bit of a wild piece of technology. Don't forget to mark as answered, have a good day!

**Reply by shellking4** [reply]
Thank you very much 🙂️ @icyJoseph  Have a good day too.

**Comment by shellking4** [comment]
For those who will come here looking for the quick answer. it's here https://github.com/vercel/next.js/discussions/37361#discussioncomment-2869895 given by @icyJoseph
